github/client.go

```diff
@@ -65,12 +66,18 @@ type Client interface {
 }
 
 // New creates a new GitHub client.
-func New(token, endpoint string, opts ...ClientOption) Client {
+func New(token, endpoint string, proxy string, opts ...ClientOption) Client {
 	cli := &http.Client{Transport: &http.Transport{
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: endpoint != "https://api.github.com",
 		},
 	}}
+	if proxy != "" {
+		proxyURL, _ := url.Parse(proxy)
+		cli = &http.Client{Transport: &http.Transport{
+			Proxy: http.ProxyURL(proxyURL),
+		}}
```

Comment:
```suggestion
		proxyURL, err := url.Parse(proxy)
		if err != nil {
			panic(err)
		}
		cli.Transport.(*http.Transport).Proxy = http.ProxyURL(proxyURL)
```
- panic is not good but it is better than ignoring errors of misconfiguration.
- Do not ignore the transport above. It is ok to cast to `*http.Transport` here.